source/common/http/http2/codec_impl.cc

```diff
@@ -2128,6 +2128,18 @@ int ServerConnectionImpl::onHeader(const nghttp2_frame* frame, HeaderString&& na
   // For a server connection, we should never get push promise frames.
   ASSERT(frame->hd.type == NGHTTP2_HEADERS);
   ASSERT(frame->headers.cat == NGHTTP2_HCAT_REQUEST || frame->headers.cat == NGHTTP2_HCAT_HEADERS);
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.http2_discard_host_header")) {
+    StreamImpl* stream = getStreamUnchecked(frame->hd.stream_id);
+    if (stream && name == static_cast<absl::string_view>(Http::Headers::get().HostLegacy)) {
```

Comment:
nit, maybe move this check above the line 2131 https://github.com/envoyproxy/envoy/blob/518c58247f51e42aec515f43be0f8f234db9458f/source/common/http/http2/codec_impl.cc#L1474-L1484

although not sure that check still available for now.